internal-packages/run-engine/src/run-queue/index.ts

```diff
@@ -825,19 +946,286 @@ export class RunQueue {
     this.subscriber.on("message", this.handleRedriveMessage.bind(this));
   }
 
-  async #callEnqueueMessage(message: OutputPayload, masterQueues: string[]) {
+  /**
+   * Debug lua scripts by publishing to this channel
+   *
+   * @example
+   *
+   * ```lua
+   * redis.call("PUBLISH", "runqueue:lua:debug", "workerQueueKey: " .. workerQueueKey .. " messageKeyValue -> " .. tostring(messageKeyValue))
+   * ```
+   */
+  async #setupLuaLogSubscriber() {
+    this.luaDebugSubscriber.subscribe("runqueue:lua:debug", (err) => {
+      if (err) {
+        this.logger.error(`Failed to subscribe to runqueue:lua:debug`, { error: err });
+      } else {
+        this.logger.log(`Subscribed to runqueue:lua:debug`);
+      }
+    });
+
+    this.luaDebugSubscriber.on("message", (_channel, msg) => {
+      this.logger.debug("runqueue lua debug", { msg });
+    });
+  }
+
+  #startMasterQueueConsumers() {
+    if (this.options.masterQueueConsumersDisabled) {
+      this.logger.debug("Master queue consumers disabled");
+
+      return;
+    }
+
+    for (let i = 0; i < this.shardCount; i++) {
+      this.logger.debug(`Starting master queue consumer ${i}`);
+      // We will start a consumer for each shard
+      this.#startMasterQueueConsumer(i).catch((err) => {
+        this.logger.error(`Failed to start master queue consumer ${i}`, { error: err });
+      });
+    }
+
+    this.logger.debug(`Started ${this.shardCount} master queue consumers`);
+  }
+
+  async #startMasterQueueConsumer(shard: number) {
+    let lastProcessedAt = Date.now();
+    let processedCount = 0;
+
+    const consumerId = nanoid();
+
+    try {
+      for await (const _ of setInterval(500, null, { signal: this.abortController.signal })) {
+        this.logger.verbose(`Processing master queue shard ${shard}`, {
+          processedCount,
+          lastProcessedAt,
+          service: this.name,
+          shard,
+          consumerId,
+        });
+
+        const now = performance.now();
+
+        const results = await this.#processMasterQueueShard(shard, consumerId);
```

Comment:
@ericallam are you very sure this function can't throw? If it does then the consumer will stop until we we restart the service. I'd use `tryCatch` here so it protects against any future changes too.